### PR TITLE
Brain-Mech fix, MMI Runtime fix, Mecha Wall Punching Fix

### PR DESCRIPTION
**Pull request by Mechoid**
Allows MMI's to be placed into Mecha once more.
-MMI's can use their own radio inside the exosuit.
-Exosuit operation keycodes must be set for the Brainmech after installation, to allow the Exosuit to interact with airlocks.
-MMI can eject.
-MMI's can't be used to unlock a DNA-locked mech, unless it was originally their lock.
MMI's given a check for their relay move to make sure the rig they detect is, in fact, a rig; Runtime fixed here.
Mechs are able to punch walls down again.

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -829,6 +839,70 @@
...
+		else
+			to_chat(user,"Occupant detected.")
+	else
+		to_chat(user,"You stop attempting to install the brain.")
+	return 0
+
+/obj/mecha/proc/mmi_moved_inside(var/obj/item/device/mmi/mmi_as_oc as obj,mob/user as mob)
+	if(mmi_as_oc && user in range(1))
+		if(!mmi_as_oc.brainmob || !mmi_as_oc.brainmob.client)
+			user << "Consciousness matrix not detected."
```
Comment (Anewbe) [inline code/game/mecha/mecha.dm]: to_chat(), if possible?

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -829,6 +839,70 @@
...
+			return 0
+		else if(mmi_as_oc.brainmob.stat)
+			user << "Beta-rhythm below acceptable level."
```
Comment (Anewbe) [inline code/game/mecha/mecha.dm]: to_chat(), if possible?

**Comment by Anewbe** [comment]
@Mechoid Confirming here that this is done being worked on?

**Comment by Mechoid** [comment]
Confirmed, unless you spot something I missed.

**Pull request merged**